x/auth/vesting/types/vesting_account.go

```diff
@@ -284,17 +283,52 @@ func (cva ContinuousVestingAccount) GetStartTime() int64 {
 // Validate checks for errors on the account fields
 func (cva ContinuousVestingAccount) Validate() error {
 	if cva.GetStartTime() >= cva.GetEndTime() {
-		return errors.New("vesting start-time cannot be before end-time")
+		return errors.New("vesting start-time cannot be before or equal to end-time")
 	}
 
 	return cva.BaseVestingAccount.Validate()
 }
 
 // UpdateSchedule updates the vesting schedule for a continuous vesting account.
-// It delegates to the base vesting account implementation and adjusts end time if needed.
-func (cva *ContinuousVestingAccount) UpdateSchedule(amount sdk.Coins) error {
-	if err := cva.BaseVestingAccount.UpdateSchedule(amount); err != nil {
-		return err
+// It calculates the proportion of the vesting amount based on the current vesting progress
+// and adds only that portion to the OriginalVesting amount.
+func (cva *ContinuousVestingAccount) UpdateSchedule(blockTime time.Time, amount sdk.Coins) error {
+	originalVesting := cva.BaseVestingAccount.GetOriginalVesting()
+	vestedCoins := cva.GetVestedCoins(blockTime)
+
+	var amountToAdd sdk.Coins
+
+	for _, coin := range amount {
+		origCoin := originalVesting.AmountOf(coin.Denom)
+		if origCoin.IsZero() {
+			// If the denomination wasn't originally vesting, skip it.
+			// Do not add new denominations to the original vesting schedule during an update.
+			continue
+		}
+
+		vestedCoin := vestedCoins.AmountOf(coin.Denom)
+
+		// Calculate the fraction that has already vested
+		vestedRatio := math.LegacyNewDecFromInt(vestedCoin).Quo(math.LegacyNewDecFromInt(origCoin))
+
+		// Calculate the fraction that is still vesting (1 - vestedRatio)
+		vestingRatio := math.LegacyOneDec().Sub(vestedRatio)
+		if vestingRatio.IsNegative() {
+			// Clamp to zero if somehow vestedRatio > 1
+			vestingRatio = math.LegacyZeroDec()
+		}
+
+		// Scale the input amount by the vesting fraction
+		addAmt := math.LegacyNewDecFromInt(coin.Amount).Mul(vestingRatio).RoundInt()
+
+		if !addAmt.IsZero() {
+			amountToAdd = amountToAdd.Add(sdk.NewCoin(coin.Denom, addAmt))
+		}
```

Comment:
This conditional seems unnecessary

```suggestion
		amountToAdd = amountToAdd.Add(sdk.NewCoin(coin.Denom, addAmt))
```

Reply:
saves a few cpu cycles but thats about it. ill remove